### PR TITLE
Init. fs container: make cp more resilient

### LINKED ISSUE #4877: Init container elastic-internal-init-filesystem should be idempotent
The `elastic-internal-init-filesystem` init container may be restarted. When it happens the container is likely to fail because the files in the `/usr/share/elasticsearch/bin` directory have the following  permissions:

```
dr-xr-xr-x. 2 elasticsearch root     4096 Jul 29 20:54 .
drwxrwxr-x. 1 elasticsearch root       20 Jul 29 20:57 ..
-r-xr-xr-x. 1 elasticsearch root     2896 Jul 29 20:47 elasticsearch
...
```

The lack of write permission (`r-xr-xr-x`) prevents the script to overwrite the destination files if they already exist:

```
'/usr/share/elasticsearch/bin/elasticsearch' -> '/mnt/elastic-internal/elasticsearch-bin-local/elasticsearch'
cp: cannot create regular file '/mnt/elastic-internal/elasticsearch-bin-local/elasticsearch': Permission denied
```

I think a workaround would be to add the `--force` flag to the `cp` command here: https://github.com/elastic/cloud-on-k8s/blob/f52ca4700f56238c56186f0fc751ada563cf6dda/pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go#L108-L109

Note that we had a [similar situation in the past](https://github.com/elastic/cloud-on-k8s/pull/3295), but only the case of the keystore was addressed.

### PR DESCRIPTION
**Pull request by mikechengwei**
Resolves #4877.

### CODE REVIEW

**Review by barkbay — COMMENTED** [review]
Thanks for the PR. Could you also adjust the unit test here: https://github.com/elastic/cloud-on-k8s/blob/0a032599fbba54208dbecce6a94a33533925d7d9/pkg/controller/elasticsearch/initcontainer/prepare_fs_script_test.go#L30-L34

Thanks !

For the record, as it is hard to reproduce the problem, I added the following code at the end of the script to do some manual testing and simulate errors:

```diff
diff --git a/pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go b/pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go
index 55a209d79..43cfc4c2e 100644
--- a/pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go
+++ b/pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go
@@ -169,6 +169,14 @@ var scriptTemplate = template.Must(template.New("").Parse(
        #         End        #
        ######################
 
+    while [ ! -f /tmp/wait ]; do sleep 5; done
+       value=$(cat /tmp/wait)
+    echo "Exit value is $value"
+    if [[ "$value" > 0 ]]; then
+               echo "Simulating error"
+               exit $value
+    fi
+
        echo "Init script successful"
        echo "Script duration: $(duration $script_start) sec."
 `))
```

File: pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ var scriptTemplate = template.Must(template.New("").Parse(
 			echo "Empty dir {{.ContainerMountPath}}"
 		else
 			echo "Copying {{.ContainerMountPath}}/* to {{.InitContainerMountPath}}/"
-			cp -av {{.ContainerMountPath}}/* {{.InitContainerMountPath}}/
+			yes | cp -avf {{.ContainerMountPath}}/* {{.InitContainerMountPath}}/
```
[the remark this answers is not in the document]
Comment (mikechengwei) [reply pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go]: Hi,I can’t understand below code mean.
```
while [ ! -f /tmp/wait ]; do sleep 5; done
       value=$(cat /tmp/wait)
    echo "Exit value is $value"
    if [[ "$value" > 0 ]]; then
               echo "Simulating error"
               exit $value
    fi
```
Can you explain this? @barkbay

File: pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go [hunk 2]
```diff
@@ -106,7 +106,8 @@ var scriptTemplate = template.Must(template.New("").Parse(
 			echo "Empty dir {{.ContainerMountPath}}"
 		else
 			echo "Copying {{.ContainerMountPath}}/* to {{.InitContainerMountPath}}/"
-			cp -av {{.ContainerMountPath}}/* {{.InitContainerMountPath}}/
+			# we want the init container to be idempotent and not fail when executed more than once,so cp command use -f args.
```
Comment (barkbay) [inline pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go:109]:
```suggestion
			# Use "yes" and "-f" as we want the init container to be idempotent and not to fail when executed more than once.
```

**Comment by barkbay** [comment]
jenkins test this please

**Comment by barkbay** [comment]
Jenkins test this please

```
java.lang.IllegalStateException: JENKINS-37121: something already locked /var/lib/jenkins/workspace/cloud-on-k8s-pr
```

**Comment by barkbay** [comment]
Jenkins test this please

https://devops-ci.elastic.co/job/cloud-on-k8s-pr/7169/ failed because of the exact same error as above but `devops-ci` pipeline is considered as ✅  🤔 

image

**Review by barkbay — APPROVED** [review]
lgtm

File: pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikechengwei) [reply pkg/controller/elasticsearch/initcontainer/prepare_fs_script.go]: I know what you mean, I have a question why is the `tmp/wait` directory , the init container execute cp command overwrite  initContainerPath.Should the test file be created under initContainerPath? @barkbay

**Pull request merged**